### ISSUE TITLE
Issues with certificate expiration rules

### ISSUE THREAD

**Issue by prontog** [samber/awesome-prometheus-alerts#327]
Hello and thanks for this great collection!

I would like to share a couple issues with the certificate expiration rules in the [blackbox](https://awesome-prometheus-alerts.grep.to/rules#blackbox) collection:

1.  The main one is [BlackboxSslCertificateExpired](https://awesome-prometheus-alerts.grep.to/rules#rule-blackbox-1-7). This one uses the `probe_ssl_earliest_cert_expiry` metric. The thing is that after a certificate has expired, *blackbox-exporter* stops exposing this metric. The best thing I could come up as a replacement is `probe_http_ssl == 0 and probe_success == 0` but it definitely is not robust since it will fire if the monitored target is unreachable.
2. The other issue is with the two [BlackboxSslCertificateWillExpireSoon](https://awesome-prometheus-alerts.grep.to/rules#rule-blackbox-1-5) rules. These are sensitive to probe failures and can sometimes flap. Imagine a target, with a cert that is close to expiration, that once in a while is slow and make the probe to timeout. Then the `probe_ssl_earliest_cert_expiry` will disappear and a firing alert with resolve. On the next successful probe the metric will reappear and the alert will fire again. To mitigate this we can use [last_over_time](https://prometheus.io/docs/prometheus/latest/querying/functions/#aggregation_over_time) such as: `last_over_time(probe_ssl_earliest_cert_expiry[10m]`.

I can create a PR if you want.

**Comment by samber** [comment]
Yes please, create a PR ;)

**Comment by prontog** [comment]
Happy New Year!  I wll try to create a PR today. 

Regarding the first issue, I discovered that the `probe_ssl_earliest_cert_expiry` metric is exposed for expired certificates if the following is added to the *blackbox-exporter* module:

```
tls_config:
        insecure_skip_verify: true
```

So I think I will simply add a comment stating this. Haven't found any good workaround using other metrics to deduce that a certificate has expired.

**Comment by prontog** [comment]
Another issue I have noticed with the [BlackboxSslCertificateWillExpireSoon](https://awesome-prometheus-alerts.grep.to/rules#rule-blackbox-1-5) is that the **30** days threshold is equal to the Let's Encrypt renewal recommendation window and is used by many of the ACME clients (including [certbot](https://eff-certbot.readthedocs.io/en/stable/using.html#renewing-certificates)). The thing is that it is very often used on a **weekly** scheduled job (again used by the `certbot` snap package) which means that it is possible for this alert to fire even if nothing is wrong.  We could change it to **20** days. This is what *Let's Encrypt* use in their [email notifications](https://letsencrypt.org/docs/expiration-emails/).

There is another issue with the thresholds. If you don't add `inhibit_rules`, then when a certificate has expired you will get three notifications. One for each of the "< 30", "< 3" and "< 0" thresholds. By adding a lower bound in the first two rules such as "3 <= metric < 30" and "0 <= metric < 3" you do not need *inhibit_rules* and there is no overlap between the alerts.

Finally, all three expiration-related alerts use *seconds* in their expressions which makes the value in the description also in seconds (and using floating numbers). If instead we used *days* in the expression the alert notification would be easier to read and validate.

I apologize for not mentioning all issues together.

I will create a PR with all of these changes. If you think that they are too many, I will create another one with the sub.
